### PR TITLE
Silence gcc-10 warnings about invalid array accesses

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: include/deal.II/base/tensor.h [hunk 1]
```diff
@@ -2391,9 +2391,14 @@ DEAL_II_CONSTEXPR inline DEAL_II_ALWAYS_INLINE
 
   Tensor<1, dim, typename ProductType<Number1, Number2>::type> result;
 
-  result[0] = src1[1] * src2[2] - src1[2] * src2[1];
-  result[1] = src1[2] * src2[0] - src1[0] * src2[2];
-  result[2] = src1[0] * src2[1] - src1[1] * src2[0];
+  // avoid compiler warnings
+  constexpr int s0 = 0 % dim;
+  constexpr int s1 = 1 % dim;
+  constexpr int s2 = 2 % dim;
```
Comment (bangerth) [inline include/deal.II/base/tensor.h:2397]: Ugh, not pretty. Alternatively, you could guard this block with a `if constexpr (dim == 3)` and make the `constexpr` conditional on C++17. Or maybe we just don't care and just go with your modification.

File: source/fe/fe_bernardi_raugel.cc [hunk 2]
```diff
@@ -177,7 +177,6 @@ FE_BernardiRaugel<dim>::initialize_support_points()
     }
 }
 
-template class FE_BernardiRaugel<1>;
```
Comment (bangerth) [inline source/fe/fe_bernardi_raugel.cc:180]: What happens?

Comment (masterleinad) [reply source/fe/fe_bernardi_raugel.cc:180]: It's the same. We unconditionally access array element 1. AFAICT, it doesn't really make sense to use this finite element for 1D anyway.

Comment (bangerth) [reply source/fe/fe_bernardi_raugel.cc:180]: I guess that makes sense. @GrahamBenHarper -- FYI.

Comment (GrahamBenHarper) [reply source/fe/fe_bernardi_raugel.cc:180]: That's fine. There is no 1D Bernardi-Raugel element. :)

**Comment by masterleinad** [comment]
The CI failure is spurious again.

**Pull request merged**